network.go

```diff
@@ -96,12 +96,19 @@ func netw(addr net.IP) string {
 	return IPV4
 }
 
-// fileExists checks if a file exists and is not a directory before we
-// try using it to prevent further errors.
-func fileExists(filename string) bool {
+// checks if the socket file exists and unlinks if it isn't a directory
+// this prevents errors from trying to use an already existing socket
+func unlinkExisting(filename string) error {
 	info, err := os.Stat(filename)
-	if os.IsNotExist(err) {
-		return false
+
```

Comment:
fist, you should check err for nil. If error is not nil -> return
if not nil -> and it is os.ErrNotExists (test should be added for this case) -> check for info != nil and info.isDir(). Needs to be checked, but I think, that you may use syscall.Unlink right after check for error != nil and skip all other checks.